src/app/features/home/components/search-panel/search-panel.component.ts

```diff
@@ -1,8 +1,66 @@
-import { Component } from '@angular/core';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { AbstractControl, FormBuilder, FormGroup } from '@angular/forms';
+import { ICard } from '../../../../core/models/cards';
+import { CardService } from '../../services/card.service';
+import { dropdownNames } from '../../../../../assets/data/dropdownStates';
+import { DropDownNames } from '../../../../core/models/dropDownNames';
+import { FilterService } from '../../services/filter.service';
 
 @Component({
   selector: 'app-search-panel',
   templateUrl: './search-panel.component.html',
   styleUrls: ['./search-panel.component.scss'],
 })
-export class SearchPanelComponent { }
+export class SearchPanelComponent implements OnInit {
+  @Input() public cards: ICard[];
+
+  @Output() public filtered: EventEmitter<ICard[]> = new EventEmitter<ICard[]>();
+
+  public searchingForm: FormGroup;
+
+  public filteredCards: ICard[];
+
+  public dropdownNames: DropDownNames = dropdownNames;
+
+  constructor(
+    private readonly formBuilder: FormBuilder,
+    private readonly cardService: CardService,
+    private readonly filterService: FilterService,
+  ) { }
+
+  ngOnInit() {
+    this.searchingForm = this.formBuilder.group({
+      search: [''],
+      dropdownType: [''],
+      dropdownPrice: [''],
+      dropdownRooms: [''],
+      dropdownRegion: [''],
+
+    });
+    this.formControlsSetInitialValues();
+  }
+
+  public get formControls(): { [key: string]: AbstractControl } {
+    return this.searchingForm.controls;
+  }
+
+  private formControlsSetInitialValues(): void {
+    this.formControls.dropdownType.setValue(this.dropdownNames.type[0]);
+    this.formControls.dropdownPrice.setValue(this.dropdownNames.price[0]);
+    this.formControls.dropdownRooms.setValue(this.dropdownNames.rooms[0]);
+    this.formControls.dropdownRegion.setValue(this.dropdownNames.region[0]);
+  }
+
+  public onSearch(): void {
+    this.filterService.show();
+    this.filteredCards = this.filterService.filterCards(this.cards, this.formControls);
+    this.filtered.emit(this.filteredCards);
+  }
+
+  public onReset(): void {
+    this.searchingForm.reset();
+    this.formControlsSetInitialValues();
```

Comment:
Is it necessary?
As far as I remember, reset should set the initial value to your group. So you need to install the initial value during creating inside ngOnInit

Reply:
reset don't set initial value
Firstly, I thought the same, but without calling a method formControlsSetInitialValues() resetting of dropdowns form  doesn't work